src/commonMain/kotlin/com/charleskorn/kaml/Annotations.kt

```diff
@@ -32,3 +33,27 @@ import kotlinx.serialization.SerialInfo
 public annotation class YamlComment(
     vararg val lines: String,
 )
+
+/**
+ * Write a String value if it is a single line in the specified ScalarStyle.
+ * This overrides the value specified in the [YamlConfiguration].
+ */
+@OptIn(ExperimentalSerializationApi::class)
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.BINARY)
+@SerialInfo
+public annotation class YamlWriteSingleLineStringUsingScalarStyle(
+    val scalarStyle: ScalarStyle,
```

Comment:
issue: the SnakeKMP dependency is specified as `implementation`. This means the end user won't be able to use `ScalarStyle` without explicitly adding the SnakeKMP into their dependencies. We probably should create a new enumeration in this library that the end user could use.
I am not sure how aliases work but if we can create an alias for `ScalarStyle` and user would be able to import it without having SnakeKMP explicitly specified in their dependency, that could be an alternative to the suggestion above.

Reply:
When I look at the YamlOutput code I find this `ScalarStyle` class to be used all over the place. How can a consumer of this library use it without using this?

Reply:
Very easy: usage is encapsulated inside YamlOutput class. You can say it is implementation details. The end user doesn't have to know about them at all.

This is all about compile and runtime dependencies. The first group is available during compilation (and usually in runtime). The second - during execution when you run the program.
The SnakeKMP is both compile and runtime dependency for this library. But it will be only a runtime dependency for those who consumes kaml. Meaning, it won't be available at compilation time. And because of that, the end user won't be able to use this class in their code - it won't simply compile.

This [link](https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_recognizing_dependencies) contains additional information about api and implementation dependencies

Reply:
So in cases where a class is annotated with this but the serialization using Kaml is not actually not executed: an application can exist without the SnakeKMP dependency.
Ok, I find this a rare case but in theory this can exist.

I'll have a look at fixing this.

Reply:
Not really, the SnakeKMP is a runtime dependency for the end user. The library consumer will have it in their **runtime** classpath. But in order to use the ScalarStyle class in their code the user must have it in **compile** classpath.
Sorry, if my explanation doesn't make it clear. I hope you could find some time to take a look at the link I shared in the previous comment - it should make everything clear.